packages/gafl-webapp-service/src/server.js

```diff
@@ -60,34 +62,74 @@ const createServer = options => {
 // This is a hash of the inline script at line 31 of the GDS template. It is added to the CSP to except the in-line
 // script. It needs the quotes.
 const scriptHash = "'sha256-+6WnXIl4mbFTCARd8N3COQmT3bJJmo32N8q8ZSQAIcU='"
-const plugIns = [
-  Inert,
-  Vision,
-  Scooter,
-  {
-    plugin: Blankie,
-    options: {
-      /*
-       * This defines the content security policy - which is as restrictive as possible
-       * It must allow web-fonts from 'fonts.gstatic.com'
-       */
-      fontSrc: ['self', 'fonts.gstatic.com', 'data:'],
-      scriptSrc: [scriptHash],
-      generateNonces: true
-    }
-  },
-  {
-    plugin: Crumb,
-    options: {
-      key: process.env.CSRF_TOKEN_COOKIE_NAME || CSRF_TOKEN_COOKIE_NAME_DEFAULT,
-      cookieOptions: {
-        isSecure: process.env.NODE_ENV !== 'development',
-        isHttpOnly: process.env.NODE_ENV !== 'development'
-      },
-      logUnauthorized: true
+const getSessionCookieName = () => process.env.SESSION_COOKIE_NAME || SESSION_COOKIE_NAME_DEFAULT
+const getPlugIns = () => {
+  const plugins = [
+    Inert,
+    Vision,
+    Scooter,
+    {
+      plugin: Blankie,
+      options: {
+        /*
+        * This defines the content security policy - which is as restrictive as possible
+        * It must allow web-fonts from 'fonts.gstatic.com'
+        */
+        fontSrc: ['self', 'fonts.gstatic.com', 'data:'],
+        scriptSrc: [scriptHash],
+        generateNonces: true
+      }
+    },
+    {
+      plugin: Crumb,
+      options: {
+        key: process.env.CSRF_TOKEN_COOKIE_NAME || CSRF_TOKEN_COOKIE_NAME_DEFAULT,
+        cookieOptions: {
+          isSecure: process.env.NODE_ENV !== 'development',
+          isHttpOnly: process.env.NODE_ENV !== 'development'
+        },
+        logUnauthorized: true
+      }
     }
+  ]
+  if (process.env.ANALYTICS_PRIMARY_PROPERTY && process.env.ANALYTICS_EXGOV_PROPERTY) {
+    plugins.push({
+      plugin: HapiGapi,
+      options: {
+        propertySettings: [
+          {
+            id: process.env.ANALYTICS_PRIMARY_PROPERTY,
+            hitTypes: ['pageview', 'event', 'ecommerce']
+          },
+          {
+            id: process.env.ANALYTICS_EXGOV_PROPERTY,
+            hitTypes: ['pageview']
+          }
+        ],
+        sessionIdProducer: request => useSessionCookie(request) ? request.state[getSessionCookieName()].id : null,
```

Comment:
It might be neater to expose the id on the request object: add a getId() function to the cache decorator return object and simply reference it ...
```suggestion
        sessionIdProducer: request => request.cache().getId
```
and then you don't need useSessionCookie() in server.